src/DurableTask.Netherite/Scaling/AzureTableLoadPublisher.cs

```diff
@@ -38,7 +38,12 @@ await foreach (var e in this.table.QueryAsync<PartitionInfoEntity>(x => x.Partit
                 }
                 if (tableBatch.Count > 0)
                 {
-                    await this.table.SubmitTransactionAsync(tableBatch, cancellationToken).ConfigureAwait(false);
+                    var response = await this.table.SubmitTransactionAsync(tableBatch, cancellationToken).ConfigureAwait(false);
+
+                    foreach(var responseItem in response.Value)
+                    {
+                        responseItem.Dispose();
+                    }
```

Comment:
similar to your changes elsewhere, could we replace this with a `using` statement?

Reply:
I don't have any great ideas here, unfortunately. My best suggestion would be to consider getting the IEnumerator or that IEnumerable (through GetEnumerator) and to simply call Dispose on that entire Enumerator (IEnumerator implements IDisposable) object in one go.

That would turn it into a 1 liner, but even then I don't think it's "better" than the explicit for loop. So yeah, no "great" ideas, just surprised there's not a more elegant alternative.

Reply:
I don't see a requirement that IEnumerator<T>.Dispose call Dispose() on its elements (see https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.ienumerator-1?view=net-8.0#notes-to-implementers); IEnumerator<T>.Dispose() is for things necessary to generate the sequence (connections, etc.). Which makes sense, especially if the elements are dynamically generated.

If you're doing this a lot you could write an extension for it.

Reply:
> just surprised there's not a more elegant alternative.

Indeed. However, I don't think calling Dispose in a loop is the problem. The problem is that the whole idea of the programmer being responsible for calling `Dispose` on all `IDisposable` objects at the correct time is very tedious and error prone to begin with, and the `using` clause is just a bit of syntactic sugar to hide the bitter taste, and it will only work for the 'easy cases' where the object lifetime is statically scoped.